src/main/java/no/unit/nva/cristin/projects/CristinApiClient.java

```diff
@@ -105,26 +106,37 @@ public ProjectsWrapper queryCristinProjectsIntoWrapperObjectWithAdditionalMetada
         throws IOException, URISyntaxException {
 
         long startRequestTime = System.currentTimeMillis();
-        List<CristinProject> enrichedProjects = queryAndEnrichProjects(requestQueryParams);
+        List<NvaProject> nvaProjects = importProjectsFromCristin(requestQueryParams);
         long endRequestTime = System.currentTimeMillis();
 
         ProjectsWrapper projectsWrapper = new ProjectsWrapper();
 
-        String requestQueryParamsAsString = String.format(TITLE_AND_LANGUAGE_QUERY_PARAM_PLACEHOLDER,
-            requestQueryParams.get(TITLE), requestQueryParams.get(LANGUAGE));
-
-        projectsWrapper.setId(buildUri(BASE_URL, QUESTION_MARK + requestQueryParamsAsString));
+        projectsWrapper.setId(buildUri(BASE_URL, QUESTION_MARK + titleAndLanguageQueryString(requestQueryParams)));
         projectsWrapper.setSize(0); // TODO: NP-2385: X-Total-Count header from Cristin response
-        projectsWrapper.setSearchString(requestQueryParamsAsString);
+        projectsWrapper.setSearchString(titleAndLanguageQueryString(requestQueryParams));
         projectsWrapper.setProcessingTime(calculateProcessingTime(startRequestTime, endRequestTime));
         // TODO: NP-2385: Use Link header / Pagination data from Cristin response in the next two values
         projectsWrapper.setFirstRecord(0);
         projectsWrapper.setNextResults(null); // TODO: Change to URI
-        projectsWrapper.setHits(transformCristinProjectsToNvaProjects(enrichedProjects));
+        projectsWrapper.setHits(nvaProjects);
 
         return projectsWrapper;
     }
 
+    private String titleAndLanguageQueryString(Map<String, String> requestQueryParams) {
```

Comment:
ould have reused UriUtils.queryParameters but i needed to maintain order of entries for unit testing

Reply:
I find this situation quite fragile. Is there a reason why the query parameters should be in a specific order? I think the solution to the problem is to normalize the URI by ordering the query parameters alphabetically. 

The method UriUtils.formatQueryParameters should then change to 
```
  private static String formatQueryParameters(Map<String, String> queryParams) {
        return queryParams.entrySet().stream()
                   .sorted(Entry.comparingByKey())
                   .map(entry -> String.format(PARAMETER_KEY_VALUE_PAIR_TEMPLATE, entry.getKey(), entry.getValue()))
                   .collect(Collectors.joining(PARAMETER_DELIMITER));
    }
```
The resource files should be corrected as well.

Reply:
About URI normalization: https://en.wikipedia.org/wiki/URI_normalization